src/main/java/org/jvnet/hudson/test/TailLog.java

```diff
@@ -94,7 +99,45 @@ public TailLog withColor(PrefixedOutputStream.AnsiColor color) {
      * @param job a {@link Job#getFullName}
      */
     public TailLog(File buildDirectory, String job, int number) {
-        tailer = Tailer.create(new File(buildDirectory, "log"), new TailerListenerAdapter() {
+        var log = buildDirectory.toPath().resolve("log");
+        tailer = Tailer.builder().setDelayDuration(Duration.ofMillis(50)).setTailable(new Tailer.Tailable() {
+            // like TailablePath
+            @Override public long size() throws IOException {
+                return Files.size(log);
+            }
+            @Override public FileTime lastModifiedFileTime() throws IOException {
+                return Files.getLastModifiedTime(log);
+            }
+            @Override public boolean isNewer(FileTime fileTime) throws IOException {
+                return Files.getLastModifiedTime(log).compareTo(fileTime) > 0;
+            }
+            @Override public Tailer.RandomAccessResourceBridge getRandomAccess(String mode) throws FileNotFoundException {
+                if (!Files.isRegularFile(log)) {
+                    throw new FileNotFoundException(log.toString());
+                }
+                return new Tailer.RandomAccessResourceBridge() {
+                    long ptr;
+                    @Override public long getPointer() throws IOException {
+                        return ptr;
+                    }
+                    @Override public void seek(long pos) throws IOException {
+                        ptr = pos;
+                    }
+                    @Override public int read(byte[] b) throws IOException {
+                        // Unlike RandomAccessFileBridge, not sensitive to file handle:
+                        try (var is = Files.newInputStream(log)) {
```

Comment:
Is this correct? If the log file gets recreated, shouldn't `ptr` be reset ?

Reply:
No, because `ptr` refers to how much content has already been printed; we do not want to restart at the top of the log file. See the associated test case, which prints 0–19 in order and then finishes.

Reply:
Ok so even if the file is recreated, you are sure it contains all the previous content already read.

Reply:
Yes that is the scenario being tolerated here, since this utility is designed to work on build logs, which are append-only, never rotated.